File: Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        for(int i = 1; i < nums.size(); i++){
            nums[i] = max(nums[i], nums[i] + nums[i - 1]);
        }
        
        return *max_element(nums.begin(), nums.end());
    }
};

File: Maximum Subarray.js
const maxSubArray = (nums) => {
    const dp = Array.from({length: nums.length}, () => 0);
    let answer = -1000000;
    
    dp[0] = nums[0];
    
    for(let i = 0; i < nums.length; i++){
        dp[i] = nums[i] < dp[i - 1] + nums[i] ? dp[i - 1] + nums[i] : nums[i];
        answer = answer < dp[i] ? dp[i] : answer;
    }
    
    return answer;
};